src/main/java/wis/Ui.java

```diff
@@ -0,0 +1,167 @@
+    public boolean parseInput(TaskList tasks) {
+        String[] words = input.split(" ");
+        Action action = InputParser.parseAction(input, words);
+        switch (action) {
+        case NONE:
+            break;
+        case ADD_TODO:
+            addTodo(tasks);
+            break;
+        case ADD_DEADLINE:
+            addDeadline(tasks);
+            break;
+        case ADD_EVENT:
+            addEvent(tasks);
+            break;
+        case MARK:
+            mark(words, tasks);
+            break;
+        case UNMARK:
+            unmark(words, tasks);
+            break;
+        case DELETE:
+            delete(words, tasks);
+            break;
+        case LIST:
+            Printer.printActionAttach(Action.LIST, tasks);
+            break;
+        case FIND:
+            find(words, tasks);
+            break;
+        case BYE:
+            return true;
+        case INVALID:
+            WisException.actionExceptionHandler(Action.INVALID);
+            break;
+        default:
+            throw new IllegalArgumentException("Illegal action argument provided.\n");
+        }
+        return false;
+    }
+
+    private void addTodo(TaskList tasks) {
+        try {
+            Todo todo = new Todo(InputParser.parseTodo(input));
+            tasks.add(todo);
+            Printer.printActionAttach(Action.ADD_TODO, todo, tasks);
+        } catch (InputMismatchException e) {
+            WisException.actionExceptionHandler(Action.ADD_TODO);
+        }
+    }
+
+    private void addDeadline(TaskList tasks) {
+        try {
+            String[] parsedString = InputParser.parseDeadline(input);
+            Deadline deadline = new Deadline(parsedString[0],
+                    InputParser.parseDateTime(parsedString[1]));
+            tasks.add(deadline);
+            Printer.printActionAttach(Action.ADD_DEADLINE, deadline, tasks);
+        } catch (InputMismatchException e) {
+            WisException.actionExceptionHandler(Action.ADD_DEADLINE);
+        } catch (DateTimeParseException e) {
+            WisException.dateTimeExceptionHandler(e);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            WisException.dateTimeExceptionHandler(e);
+        }
+    }
+
+    private void addEvent(TaskList tasks) {
+        try {
+            String[] parsedString = InputParser.parseEvent(input);
+            Event event = new Event(parsedString[0],
+                    InputParser.parseDateTime(parsedString[1]),
+                    InputParser.parseDateTime(parsedString[2]));
+            tasks.add(event);
+            Printer.printActionAttach(Action.ADD_EVENT, event, tasks);
+        } catch (InputMismatchException e) {
+            WisException.actionExceptionHandler(Action.ADD_EVENT);
+        } catch (DateTimeParseException e) {
+            WisException.dateTimeExceptionHandler(e);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            WisException.dateTimeExceptionHandler(e);
+        }
+    }
+
```

Comment:
I like how you seperated the creation of the different events into very clear functions. Although, may I suggest that you abstract away the construction of deadline event todo into their respective contractors so that you can have a simple addTask(TaskList tasks, Task task) method to do the adding. HAHAHA paiseh for class part 😃